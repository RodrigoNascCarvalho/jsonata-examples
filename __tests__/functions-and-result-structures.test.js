const { testJsonataExpr } = require('./jsonata-helper')
const inputJson = require('example-json')

describe('Functions + Result Structures Examples', () => {
    test('FR01: Flattening of data + concat of data into string', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description.(
            Colour & ':' & 
            Width & ':' &
            Height & ':' &
            Depth & ':' &
            Weight
        )
        `)
    })
    
    test('FR02: Only return product descriptions where width is larger than height AND color is black', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description[
            Width > Height and Colour = 'Black'
        ]
        `)
    })
    
    test('FR03: Using composition of queries, we can achieve the same result as using an AND operator', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description[
            (
                Width > Height;
                Colour = 'Black';
            )
        ]`)
    })
    
    test('FR04: Only return product descriptions where width is larger than height OR color is black', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description[
            Width > Height or Colour = 'Black'
        ]
        `)
    });
    
    test('FR05: Numeric operations on data for widths and heights resulting in array with all results for each Product Description', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description.[
            Width + Height,
            Width - Height,
            Width * Height,
            Width / Height,
            Width % Height
          ]
        `)
    })
    
    test('FR06: You can use parenthesis to set precedence of operations', () => {
        testJsonataExpr(inputJson, `
        Account.Order.Product.Description.[
            (Width + Height) * Depth
        ]
        `)
    })    
})
