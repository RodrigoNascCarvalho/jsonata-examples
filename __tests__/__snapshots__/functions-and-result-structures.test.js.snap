// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functions + Result Structures Examples FR01: Flattening of data + concat of data into string 1`] = `
Array [
  "Purple:300:200:210:0.75",
  "Orange:300:200:210:0.6",
  "Blue:300:200:210:0.75",
  "Black:30:20:210:2",
  "Yellow:300:200:210:0.75",
  "Green:30:100:210:2",
]
`;

exports[`Functions + Result Structures Examples FR02: Only return product descriptions where width is larger than height AND color is black 1`] = `
Object {
  "Colour": "Black",
  "Depth": 210,
  "Height": 20,
  "Weight": 2,
  "Width": 30,
}
`;

exports[`Functions + Result Structures Examples FR03: Using composition of queries, we can achieve the same result as using an AND operator 1`] = `
Object {
  "Colour": "Black",
  "Depth": 210,
  "Height": 20,
  "Weight": 2,
  "Width": 30,
}
`;

exports[`Functions + Result Structures Examples FR04: Only return product descriptions where width is larger than height OR color is black 1`] = `
Array [
  Object {
    "Colour": "Purple",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
  Object {
    "Colour": "Orange",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.6,
    "Width": 300,
  },
  Object {
    "Colour": "Blue",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
  Object {
    "Colour": "Black",
    "Depth": 210,
    "Height": 20,
    "Weight": 2,
    "Width": 30,
  },
  Object {
    "Colour": "Yellow",
    "Depth": 210,
    "Height": 200,
    "Weight": 0.75,
    "Width": 300,
  },
]
`;

exports[`Functions + Result Structures Examples FR05: Numeric operations on data for widths and heights resulting in array with all results for each Product Description 1`] = `
Array [
  Array [
    500,
    100,
    60000,
    1.5,
    100,
  ],
  Array [
    500,
    100,
    60000,
    1.5,
    100,
  ],
  Array [
    500,
    100,
    60000,
    1.5,
    100,
  ],
  Array [
    50,
    10,
    600,
    1.5,
    10,
  ],
  Array [
    500,
    100,
    60000,
    1.5,
    100,
  ],
  Array [
    130,
    -70,
    3000,
    0.3,
    30,
  ],
]
`;

exports[`Functions + Result Structures Examples FR06: You can use parenthesis to set precedence of operations 1`] = `
Array [
  Array [
    105000,
  ],
  Array [
    105000,
  ],
  Array [
    105000,
  ],
  Array [
    10500,
  ],
  Array [
    105000,
  ],
  Array [
    27300,
  ],
]
`;
